polygon/sync/sync.go

```diff
@@ -306,6 +313,26 @@ func (s *Sync) applyNewBlockOnTip(ctx context.Context, event EventNewBlock, ccb
 		return nil
 	}
 
+	go func() {
+		for i := range newConnectedHeaders {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				wiggle, err := s.wiggleCalculator.CalculateWiggle(ctx, newConnectedHeaders[i])
+				if err != nil {
+					s.logger.Error(
+						syncLogPrefix("failed update wiggle metrics"),
+						"err", err,
+					)
+					continue
+				}
+
+				UpdateWiggleDuration(wiggle)
+			}
+		}
+	}()
```

Comment:
It's better to explicitly pass `ctx` and `block` here so that it is easy for readers to know what is being passed to the goroutine from the calling function.